csrc/lower_validation.cpp

```diff
@@ -668,13 +707,8 @@ void fillProducerVectorizedContigRootDomains(
 
   VectorizedSetInfo& info = *it;
 
-  // info.vectorized_root_id is validated at this point to be the
-  // last concrete root domain in consumer.
-  auto consumer_root_id = info.vectorized_root_id;
-
-  auto root_id = c2p_map.at(consumer_root_id);
-
-  fillVectorizedContigRootDomains(producer_tv, contig_finder, root_id, info);
```

Comment:
This does not work with arbitrary allocation domain. The `root_id` is not necessarily inside the allocation domain of the producer, which is incorrect. We need to compute `root_id` for producer separately.

Reply:
What's exactly incorrect? Do we need to find an ID in the allocation domain? Is it what's changed in this PR?

Reply:
For example, if the producer has a root domain and has an rFactor domain different from the root domain, and the allocation domain of the producer is the root domain, then `root_id` will be an id in the rFactor domain in the producer, but we want to further traverse backward all the way towards the root domain of producer.

Reply:
And yes, this is the main issue I am trying to fix.